project_euler/problem_97/sol1.py

```diff
@@ -0,0 +1,38 @@
+"""
+The first known prime found to exceed one million digits was discovered in 1999,
+and is a Mersenne prime of the form 2**6972593 − 1; it contains exactly 2,098,960
+digits. Subsequently other Mersenne primes, of the form 2**p − 1, have been found
+which contain more digits.
+However, in 2004 there was found a massive non-Mersenne prime which contains
+2,357,207 digits: (28433 * (2 ** 7830457 + 1)).
+
+Find the last ten digits of this prime number.
+"""
+
+
+def solution(n: int = 10) -> str:
+    """
+    Returns the last n digits of NUMBER.
+    >>> solution()
+    '8739992577'
+    >>> solution(8)
+    '39992577'
+    >>> solution(1)
+    '7'
+    >>> solution(-1)
+    ''
+    >>> solution(8.3)
+    ''
+    """
+    if n < 0 or not isinstance(n, int):
+        return ""

```

Comment:
Why not raise ValueError on invalid input?

Reply:
Okay if you say so!

Reply:
CONTRIBUTING.md says so.

Reply:
Okay! 👍

Reply:
Also, please put isinstance() first and n < 0 second so that ValueError is raised on `solution("a")` instead of some other exception.